Sources/QuickCheck/ArbitraryDecodable.swift

```diff
@@ -0,0 +1,218 @@
+import Foundation
+import LCG
+import Prelude
+
+private let genInt8 = choose(Int(Int8.min)..<Int(Int8.max)).map(Int8.init)
+private let genInt16 = choose(Int(Int16.min)..<Int(Int16.max)).map(Int16.init)
+private let genInt32 = genInt.map(Int32.init)
+private let genInt64 = genInt.map(Int64.init)
+private let genUInt = choose(0..<2_000_000).map(UInt.init)
+private let genUInt8 = choose(0..<Int(UInt8.max)).map(UInt8.init)
+private let genUInt16 = choose(0..<Int(UInt16.max)).map(UInt16.init)
+private let genUInt32 = choose(0..<2_000_000).map(UInt32.init)
+private let genUInt64 = choose(0..<2_000_000).map(UInt64.init)
+private let genFloat = genDouble.map(Float.init)
+
+private final class ArbitraryDecoder: Decoder {
```

Comment:
We we could make this a `struct` (under the hood it handles the mutation for us), all the sub-containers message back to increment the seed. I'll explore in another commit, but it was feeling messy.